src/queueUI/Line.java

```diff
@@ -0,0 +1,75 @@
+package queueUI;
+
+import javax.swing.*;
+
+
+public class Line {
+
+    private Node startQueue, endQueue;
+    String queue = "";
+
+    // create constructor
+    public Line() {
+        startQueue = null;
+        endQueue = null;
+    }
+
+    // method to know if the queue is empty.
+    public boolean EmptyQueue() {
+        if (startQueue == null) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    // method to insert information in the queue
+    public void Insert(int inputData) {
+        Node recentNode = new Node();
+        recentNode.inputData = inputData;
+        recentNode.after = null;
+        if (EmptyQueue()) {
+            startQueue = recentNode;
+            endQueue = recentNode;
+        } else {
+            endQueue.after = recentNode;
+            endQueue = recentNode;
+        }
+    }
+
+    // method to extract information in the queue
+    public int Extract() {
+        if (!EmptyQueue()) {
+            int inputData = startQueue.inputData;
+            if (startQueue == endQueue) {
+                startQueue = null;
+                endQueue = null;
+            } else {
+                startQueue = startQueue.after;
+            }
+            return inputData;
+        } else {
+            return 0;
+        }
```

Comment:
We would like to use early returns to avoid nesting.
This could be re-written as:
```java
public int Extract() {
    if (EmptyQueue()) {
        return 0
    }

    int inputData = startQueue.inputData;
    if (startQueue == endQueue) {
        startQueue = null;
        endQueue = null;
    } else {
        startQueue = startQueue.after;
    }
    return inputData;
}
```